src/main/java/io/github/thebusybiscuit/slimefun4/implementation/listeners/CrashHelmetListener.java

```diff
@@ -0,0 +1,34 @@
+package io.github.thebusybiscuit.slimefun4.implementation.listeners;
+
+import io.github.thebusybiscuit.slimefun4.implementation.SlimefunPlugin;
+import me.mrCookieSlime.Slimefun.Objects.SlimefunItem.SlimefunItem;
+import org.bukkit.entity.Player;
+import org.bukkit.event.EventHandler;
+import org.bukkit.event.Listener;
+import org.bukkit.event.entity.EntityDamageEvent;
+import org.bukkit.inventory.ItemStack;
+
+import javax.annotation.Nonnull;
+
+public class CrashHelmetListener implements Listener {
+
+    public CrashHelmetListener(@Nonnull SlimefunPlugin plugin) {
+        plugin.getServer().getPluginManager().registerEvents(this, plugin);
+    }
+
+    @EventHandler
+    public void onPlayerCrash(EntityDamageEvent e) {
+        if (!(e.getEntity() instanceof Player)) return;
+        if (!(e.getCause() == EntityDamageEvent.DamageCause.FALL ||
+            e.getCause() == EntityDamageEvent.DamageCause.FLY_INTO_WALL)) return;
```

Comment:
```suggestion
        if (e.getCause() != EntityDamageEvent.DamageCause.FALL &&
            e.getCause() == EntityDamageEvent.DamageCause.FLY_INTO_WALL) return;
```

Reply:
no im trying to see if the type is not one of those 2

Reply:
Currently you only check if it's not falling or it is crasjing into the wall this would cause it to fire even when player is poisoned...

Reply:
@Seggan ^

Reply:
@LinoxGH this is supposed to fire when the players posioned. It's the return. Your solution would check only if it's crashing. I included both bc I don't know which one is the kinetic energy one.

Reply:
Oh I see now